### ISSUE TITLE
Running JNLP on HiDPI(4k) displays with scaling

### ISSUE THREAD

**Issue by al3xxx** [AdoptOpenJDK/IcedTea-Web#836]
Hello,

I have a problem with IcedTea-web executing JNLP file from browser (namely Intel rmm3 remote console Jviewer) on HiDPI display in X11 - windows are too small and fonts are just unreadable.

It's possible to workaround with setting up environment variable **GDK_SCALE=3** or **GDK_DPI_SCALE=1.5** and that works just fine while executing **/usr/bin/itweb-settings** from command line.

I tried to create simple shell wrapper as /usr/bin/javaws
```
#!/bin/sh
export GDK_SCALE=3
/usr/share/icedtea-web/bin/javaws  $@
```
but that didn't work out. Quick search on the web says that I need to pass -Dsun.java2d.uiScale=3 argument to javaws but adding it to the command line inside the wrapper didn't work for running JNLP files from web page either. 

Current workaround is to run browser with set GDK_SCALE=3, this way env variable is passed to IcedTea.

Is there any other way to make icedTea DPI aware/scale interface on HiDPI screens?

**Comment by janakmulani** [comment]
You can try specifying  [DPI related VM arguments](https://news.kynosarges.org/2019/03/24/swing-high-dpi-properties/) in your jnlp file.

**Comment by al3xxx** [comment]
No, I can not. As I mentioned, it's piece of Intel hardware that contains JNLP file. And I have about 30 of such devices. Not to mention that dialogs of the IcedTea-Web itself are too small because it's not DPI aware.

**Comment by al3xxx** [comment]
JNLP is located inside Intel firmware and out of my control. I was looking into ways to adjust IcedTea settings to be able to enlarge plugin windows as well.

**Comment by janakmulani** [comment]
Is it possible for you to try running your Jnlp with OpenWebStart?

**Comment by janakmulani** [comment]
For ITW, can you try specifying JVM DPI related arguments  using `itweb-settings.exe` 
![image](https://user-images.githubusercontent.com/655389/140656191-56a2f04a-850e-4558-a9ae-ab1c205d48af.png)

**Comment by al3xxx** [comment]
![image](https://user-images.githubusercontent.com/2903428/140657634-75f8c0f4-458b-41ae-b1fc-b24a6cba05a0.png)
This didn't work out - plugin splash and warning dialogs still small

**Comment by patris70** [comment]
@al3xxx,

use this parameter in windows registry:

javaws.exe **-J-Dsun.java2d.uiScale=3**

regards

**Comment by lherschi** [comment]
I use 
export JAVA_TOOL_OPTIONS="-Dsun.java2d.uiScale=2"
This works for all Java programs.
